### ISSUE TITLE
Long running test: SynchronizationContextCurrent_NotUsedForAsyncOperations

### ISSUE THREAD

**Issue by ViktorHofer** [dotnet/runtime#53987]
```
  Discovering: System.Xml.RW.XmlReader.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  System.Xml.RW.XmlReader.Tests (found 21 test cases)
  Starting:    System.Xml.RW.XmlReader.Tests (parallel test collections = on, max threads = 2)
   System.Xml.RW.XmlReader.Tests: [Long Running Test] 'System.Xml.Tests.AsyncReaderLateInitTests.SynchronizationContextCurrent_NotUsedForAsyncOperations', Elapsed: 00:02:11
   System.Xml.RW.XmlReader.Tests: [Long Running Test] 'System.Xml.Tests.AsyncReaderLateInitTests.SynchronizationContextCurrent_NotUsedForAsyncOperations', Elapsed: 00:04:11
   System.Xml.RW.XmlReader.Tests: [Long Running Test] 'System.Xml.Tests.AsyncReaderLateInitTests.SynchronizationContextCurrent_NotUsedForAsyncOperations', Elapsed: 00:06:11
   System.Xml.RW.XmlReader.Tests: [Long Running Test] 'System.Xml.Tests.AsyncReaderLateInitTests.SynchronizationContextCurrent_NotUsedForAsyncOperations', Elapsed: 00:08:11
   System.Xml.RW.XmlReader.Tests: [Long Running Test] 'System.Xml.Tests.AsyncReaderLateInitTests.SynchronizationContextCurrent_NotUsedForAsyncOperations', Elapsed: 00:10:11
   System.Xml.RW.XmlReader.Tests: [Long Running Test] 'System.Xml.Tests.AsyncReaderLateInitTests.SynchronizationContextCurrent_NotUsedForAsyncOperations', Elapsed: 00:12:11
   System.Xml.RW.XmlReader.Tests: [Long Running Test] 'System.Xml.Tests.AsyncReaderLateInitTests.SynchronizationContextCurrent_NotUsedForAsyncOperations', Elapsed: 00:14:11
```

Log: https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-53023-merge-09de05e127d14c2590/System.Xml.RW.XmlReader.Tests/console.8899153b.log?sv=2019-07-07&se=2021-06-30T07%3A54%3A32Z&sr=c&sp=rl&sig=4sF7Do1BpO1vQhTHcfHI7xG2DgPyABmjKhe4jrUrXrg%3D

Configuration: `net6.0-Linux-Debug-x64-Mono_release-(Centos.8.Amd64.Open)Ubuntu.1604.Amd64.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:centos-8-helix-20201229003624-c1bf759`

**Comment by akoeplinger** [comment]
First failure according to AzDO is 197cfb6 and last successful one 27baae9, so something in this range: https://github.com/dotnet/runtime/compare/27baae9...197cfb6.

**Comment by akoeplinger** [comment]
Seems to be specific to mono, will disable the test for now.

**Comment by akoeplinger** [comment]
The only commit that looks related is https://github.com/dotnet/runtime/commit/a7a2fd6543ff71cecbbfe901b81ee27a6cf428c0 by @kouvel.

/cc @lambdageek

**Comment by ViktorHofer** [comment]
Adding the disabled-test label.

**Comment by lambdageek** [comment]
Hmm.  The code to perform the adjustment depends on `GCMemoryInfo`

https://github.com/dotnet/runtime/blob/a7a2fd6543ff71cecbbfe901b81ee27a6cf428c0/src/libraries/System.Private.CoreLib/src/System/Threading/PortableThreadPool.cs#L376-L385

and it's used here:

https://github.com/dotnet/runtime/blob/a7a2fd6543ff71cecbbfe901b81ee27a6cf428c0/src/libraries/System.Private.CoreLib/src/System/Threading/PortableThreadPool.Blocking.cs#L191-L213

@kouvel how would this behave if the GCMemoryInfo data wasn't right?  @naricc did we implement all of these GCMemoryInfo properties?

**Comment by lambdageek** [comment]
@radekdoulik  I don't understand what's going on with`GCMemoryInfo.MemoryLoadBytes` here

https://github.com/dotnet/runtime/blob/93cf5df65fc13bfbc29256acc8c9c0e102a1170b/src/mono/mono/sgen/sgen-memory-governor.c#L148

https://github.com/dotnet/runtime/commit/93cf5df65fc13bfbc29256acc8c9c0e102a1170b

**Comment by radekdoulik** [comment]
> 
> 
> @radekdoulik I don't understand what's going on with`GCMemoryInfo.MemoryLoadBytes` here
> 
> https://github.com/dotnet/runtime/blob/93cf5df65fc13bfbc29256acc8c9c0e102a1170b/src/mono/mono/sgen/sgen-memory-governor.c#L148
> 
> [93cf5df](https://github.com/dotnet/runtime/commit/93cf5df65fc13bfbc29256acc8c9c0e102a1170b)

It calculates the load bytes as `total - available` size and scales it by `max_heap/physical_ram_size` ratio if `max_heap` is used.

**Comment by naricc** [comment]
@lambdageek Yes, we did implement all the  GCMemoryInfo properties. Although there is no way to garauntee they behave 100 % the same as in CoreGC, given the different garbage collectors.

**Comment by kouvel** [comment]
The GC memory info is only used as a heuristic in https://github.com/dotnet/runtime/commit/a7a2fd6543ff71cecbbfe901b81ee27a6cf428c0, when low memory is detected it would fall back to the same behavior as before the change
